run.go

```diff
@@ -67,20 +70,55 @@ func run(context *cli.Context) error {
 		return errors.New("invalid runtime config")
 	}
 
-	if err := create(context.Args().First(),
+	consolePath := context.String("console")
+
+	var wg sync.WaitGroup
+	var console Console
+	var err error
+	var consoleState *term.State
+
+	// if consolePath is /dev/ptmx or /dev/pts/ptmx
+	// means that we have to allocate a new pts
+	if consolePath == ptmxPath || consolePath == ptsPtmxPath {
+		console, err = newConsole()
+		if err != nil {
+			return err
+		}
+
+		// now consolePath is the slave pts
+		consolePath = console.Path()
+	}
+
+	if err = create(context.Args().First(),
 		context.String("bundle"),
-		context.String("console"),
+		consolePath,
 		context.String("pid-file"),
 		runtimeConfig); err != nil {
 		return err
 	}
 
+	detach := context.Bool("detach")
+
+	if !detach {
+		wg.Add(1)
+		go io.Copy(console, os.Stdin)
+		go func() {
+			defer wg.Done()
+			io.Copy(os.Stdout, console)
+		}()
```

Comment:
Why are you not copying stderr ? You should call Add(2) and make sure you're copying stderr the same way you copy stdout, right ?